### PR TITLE
[python-updates] Recurse into python313Packages

### PR DESCRIPTION
**Pull request by mweinelt**
Early prep for the 25.05 release cycle.

- Recurse into python313Packages
- Stop recursing into python311Packages
- Update the default numpy version

### CODE REVIEW

**Comment by natsukium** [comment]
Regarding numpy2, when will we update it? Since python313 package set adopts numpy2 by default, I think it would be a good idea to bump it here to avoid confusion.

I have a few fixes and here is my local hydra evaluation as of about a week ago.
I'm a little concerned about jax, but the rest looks good.

Diff:
https://github.com/NixOS/nixpkgs/compare/master...natsukium:nixpkgs:numpy2

Hydra evaluation report:
https://gist.github.com/natsukium/e4362b577d41ac8f8b7962446c0c4c44

**Comment by mweinelt** [comment]
> Regarding numpy2, when will we update it? Since python313 package set adopts numpy2 by default, I think it would be a good idea to bump it here to avoid confusion.

Let's give it a shot. I said we'd delay it until after the release, and here we are.

> I'm a little concerned about jax, but the rest looks good.

Too early to worry imo.

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/524.html

**Comment by mweinelt** [comment]
Okay, so jax fails a number of tests (the same on 3.12 and 3.13), but we can still try to update it.

```
FAILED tests/debugger_test.py::CliDebuggerTest::test_debugger_can_print_backtrace - jaxlib.xla_extension.XlaRuntimeError: INTERNAL: Generated function failed: ...
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUnique1 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUnique2 - AssertionError: Tuples differ: (2,) != (2, 1, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUnique4 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize0 - AssertionError: Tuples differ: (4,) != (1, 4)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize4 - AssertionError: Tuples differ: (4,) != (1, 4)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize5 - AssertionError: Tuples differ: (1,) != (1, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize6 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize8 - AssertionError: Tuples differ: (4,) != (1, 1, 4)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize9 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::NumpySignaturesTest::testWrappedSignaturesMatch - AssertionError: {'clip': {'np_params': ['a', 'a_min', 'a_max', 'min', 'max'...
FAILED tests/lax_numpy_reducers_test.py::JaxNumpyReducerTests::testCumulativeSum0 - AssertionError:
```

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/536.html

**Comment by dotlambda** [comment]
@teh @thoughtpolice @tscholak Can we remove six and future from torch's dependencies?

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/549.html

File: pkgs/applications/misc/octoprint/default.nix [hunk 1]
```diff
@@ -34,6 +34,7 @@ let
           flask = super.flask.overridePythonAttrs (oldAttrs: rec {
             version = "2.2.5";
             format = "setuptools";
+            pyproject = null;
```
Comment (dotlambda) [inline pkgs/applications/misc/octoprint/default.nix:37]: Why not override `build-system` instead?

Comment (mweinelt) [reply pkgs/applications/misc/octoprint/default.nix:37]: No strong opinion. Was just focused on getting rid of the eval error.

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/551.html

**Comment by doronbehar** [comment]
I iterated some of the commits related to packages I have some interest in, and they look good. Not sure if I have time to help further. Sorry.

**Pull request merged**